src/target/riscv/riscv-013.c

```diff
@@ -417,7 +424,21 @@ static void select_dmi(struct target *target)
 		select_dmi_via_bscan(target);
 		return;
 	}
-	jtag_add_ir_scan(target->tap, &select_dbus, TAP_IDLE);
+
+	uint8_t tmp_ir_dbus[4] = {0};
+	if (target->dbgbase_set) {
+		tmp_ir_dbus[0] = (target->dbgbase + DBUS) & 0xff;
+		tmp_ir_dbus[1] = ((target->dbgbase + DBUS) >> 8) & 0xff;
+		tmp_ir_dbus[2] = ((target->dbgbase + DBUS) >> 16) & 0xff;
+		tmp_ir_dbus[3] = ((target->dbgbase + DBUS) >> 24) & 0xff;
+	} else {
+		tmp_ir_dbus[0] = DBUS;
+	}
+	struct scan_field tmp_select_dbus;
+	memcpy(&tmp_select_dbus, &select_dbus, sizeof(struct scan_field));
+	tmp_select_dbus.out_value = tmp_ir_dbus;
```

Comment:
It's simpler to get rid of the global ir_dbus and just create a new one here than it is to mess around with memcpy.

Reply:
There should be an error or something in case dbgbase is larger than can be accommodated by abits. We should probably check for that in examine(), assuming that dbgbase must be set by then.

Reply:
> It's simpler to get rid of the global ir_dbus and just create a new one here than it is to mess around with memcpy.

As memcpy is for select_dbus,  I guess you mean remove both global ir_dbus and global select_dbus.
I use memcpy because I'm worried that global select_dbus may change in somewhere I haven't noticed.
I agree get rid of global and memcpy is better, I will do that

Reply:
new patch don't modify this funtion